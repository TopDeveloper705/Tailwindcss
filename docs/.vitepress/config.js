// @ts-check

const isProd = process.env.NODE_ENV === 'production'

const title = 'Windi CSS ðŸŒ¬'
const description = 'Next generation Tailwind CSS compiler.'
const site = isProd ? 'https://windicss.github.io' : 'http://localhost:3000'
const image = `${site}/banner.png`

const head = [
  ['style', {}, 'img { border-radius: 10px }' + 'h1.title { margin-left: 0.5em }'],
  ['meta', { name: 'author', content: 'Windi CSS Team' }],
  ['meta', { name: 'keywords', content: 'windi, tailwind, tailwindcss, vitejs, svelte, vue' }],

  ['link', { rel: 'icon', type: 'image/svg+xml', href: '/assets/logo.svg' }],

  ['meta', { name: 'HandheldFriendly', content: 'True' }],
  ['meta', { name: 'MobileOptimized', content: '320' }],
  ['meta', { name: 'theme-color', content: '#cc0000' }],

  ['meta', { name: 'twitter:card', content: 'summary_large_image' }],
  ['meta', { name: 'twitter:site', content: site }],
  ['meta', { name: 'twitter:title', value: title }],
  ['meta', { name: 'twitter:description', value: description }],
  ['meta', { name: 'twitter:image', content: image }],

  ['meta', { property: 'og:type', content: 'website' }],
  ['meta', { property: 'og:locale', content: 'en_US' }],
  ['meta', { property: 'og:site', content: site }],
  ['meta', { property: 'og:site_name', content: title }],
  ['meta', { property: 'og:title', content: title }],
  ['meta', { property: 'og:image', content: image }],
  ['meta', { property: 'og:description', content: description }],
]

/**
 * @type {import('vitepress').UserConfig}
 */
module.exports = {
  title: 'Windi CSS',
  description,
  head,
  themeConfig: {
    repo: 'windicss/windicss',
    logo: '/assets/logo.svg',
    docsDir: 'docs',
    docsBranch: 'main',
    editLinks: true,
    editLinkText: 'Suggest changes to this page',

    nav: [
      { text: 'Guide', link: '/guide/' },
      { text: 'Config', link: '/config/' },
      {
        text: 'Changelog',
        link: 'https://github.com/windicss/windicss/blob/main/CHANGELOG.md',
      },
    ],

    sidebar: {
      '/config/': 'auto',
      // catch-all fallback
      '/': [
        {
          text: 'Guide',
          children: [
            { text: 'Introduction', link: '/guide/introduction' },
            { text: 'Getting Started', link: '/guide/' },
            { text: 'Migration', link: '/guide/migration' },
          ],
        },
      ],
    },
  },
}
