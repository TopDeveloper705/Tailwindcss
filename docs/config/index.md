# Configuring Windi CSS

Coming soon.

In the meantime, have in mind that it supports the same options as Tailwind CSS.
