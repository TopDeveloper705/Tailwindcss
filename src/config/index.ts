export { baseConfig } from './base';
export { colors } from './colors';
export const twExclude = [
  /-hex-/,                          // disable hex color
  /-\$/,                            // disable variable
  /(rem|em|px|rpx|vh|vw|ch|ex)$/,   // disable size
  /-\d*[13579]$/,                   // disable odd number
  /-([0-9]{1,}[.][0-9]*)$/,         // disable float
  /^!/,                             // disable important utility
];
