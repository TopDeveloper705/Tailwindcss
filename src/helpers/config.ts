import type { Config } from '../interfaces';

export function defineConfig(config: Config): Config {
  return config;
}
