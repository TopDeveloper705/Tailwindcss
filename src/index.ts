export * from './processor';
export * from './utils';
export * from './config';
export * from './plugin';

module.exports = {
    'colors': require('./config/colors')
}

