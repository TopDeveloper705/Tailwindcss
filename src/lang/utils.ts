export function isSpace(str: string): boolean {
  return /\s/.test(str);
}

export function isDigit(str: string): boolean {
  return /^\d+$/.test(str);
}

export function isAlpha(str: string): boolean {
  return /\w/.test(str);
}

export function connect(strings: string[]): string {
  return strings.join(';\n') + ';';
}

export function findGroupEnd(text: string, startIndex = 0): number {
  let level = 1;
  let endBracket = searchFrom(text, '}', startIndex);
  while (endBracket !== -1) {
    const nextBracket = searchFrom(text, '{', startIndex);
    if (endBracket < nextBracket || nextBracket === -1) {
      level--;
      startIndex = endBracket + 1;
      if (level === 0) return endBracket;
    } else {
      level++;
      startIndex = nextBracket + 1;
    }
    endBracket = searchFrom(text, '}', startIndex);
  }
  return -1;
}

export function searchFrom(
  text: string,
  target: string | RegExp,
  startIndex = 0,
  endIndex?: number
): number {
  // search from partial of string
  const subText = text.substring(startIndex, endIndex);
  const relativeIndex = subText.search(target);
  return relativeIndex === -1 ? -1 : startIndex + relativeIndex;
}
