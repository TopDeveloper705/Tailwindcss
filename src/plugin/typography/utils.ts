export function uniq<T>(array: T[]): T[] {
  return Array.from(new Set(array));
}

export function castArray<T>(value: T): unknown[] {
  if (!arguments.length) {
    return [];
  }
  return Array.isArray(value) ? value : [value];
}

export function isUsableColor(color: string, values: string | {[key:string]:string}): boolean {
  return Boolean(values && typeof values === 'object' && color !== 'gray' && values[600]);
}

export const round = (num: number): string =>
  num
    .toFixed(7)
    .replace(/(\.[0-9]+?)0+$/, '$1')
    .replace(/\.0$/, '');

export const rem = (px: number): string => `${round(px / 16)}rem`;

export const em = (px: number, base: number): string => `${round(px / base)}em`;
