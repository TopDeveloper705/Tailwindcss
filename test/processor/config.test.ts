import { resolve } from 'path';
import { toType } from '../../src/utils/tools';
import { Processor } from '../../src/lib';
import { twExclude } from '../../src/config';

const configPath = resolve('./test/assets/tailwind.config.js');
const userConfig = require(configPath);

describe('Config', () => {
  const baseConfig = new Processor();

  it('dict input', () => {
    const processor = new Processor(userConfig);
    expect(processor.config('theme.screens')).toEqual(
      processor.theme('screens')
    );
    expect(processor.theme('screens')).toEqual(userConfig.theme.screens);
    expect(processor.theme('colors')).toEqual(userConfig.theme.colors);
    expect(processor.theme('colors.pink')).toEqual(
      userConfig.theme.colors.pink
    );
    expect(processor.theme('fontFamily')).toEqual(userConfig.theme.fontFamily);
    expect(processor.theme('spacing')).toEqual({
      ...(toType(baseConfig.theme('spacing'), 'object') ?? {}),
      ...userConfig.theme.extend.spacing,
    });
    expect(processor.theme('borderRadius')).toEqual({
      ...(toType(baseConfig.theme('borderRadius'), 'object') ?? {}),
      ...userConfig.theme.extend.borderRadius,
    });
  });

  it('resolveConfig should extend correctly', () => {
    const processor = new Processor({
      theme: {
        extend: {
          order: {
            lg: '44',
          },
          lineClamp: {
            sm: '4',
            lg: '9',
          },
        },
      },
    });
    expect(processor.theme('order')).toEqual({
      first: '-9999',
      last: '9999',
      none: '0',
      lg: '44',
    });
    expect(processor.theme('lineClamp')).toEqual({ sm: '4', lg: '9' });
  });

  it('resolveConfig should not overwrite theme function', () => {
    const processor = new Processor({
      theme: {
        extend: {
          width: {
            '1/7': '14%',
            '2/7': '28%',
            '3/7': '42%',
            '4/7': '57%',
            '5/7': '71%',
            '6/7': '85%',
          },
        },
      },
    });
    expect(processor.theme('width')).toEqual({
      auto: 'auto',
      px: '1px',
      xs: '20rem',
      sm: '24rem',
      md: '28rem',
      lg: '32rem',
      xl: '36rem',
      '2xl': '42rem',
      '3xl': '48rem',
      full: '100%',
      min: 'min-content',
      max: 'max-content',
      prose: '65ch',
      screen: '100vw',
      'screen-sm': '640px',
      'screen-md': '768px',
      'screen-lg': '1024px',
      'screen-xl': '1280px',
      'screen-2xl': '1536px',
      '1/7': '14%',
      '2/7': '28%',
      '3/7': '42%',
      '4/7': '57%',
      '5/7': '71%',
      '6/7': '85%',
    });
  });

  it('user theme should overwrite default theme', () => {
    const processor = new Processor({
      theme: {
        colors: {
          white: '#ffffff',
        },
      },
    });
    expect(processor.theme('colors')).toEqual({
      white: '#ffffff',
    });
  });

  it('change separator test', () => {
    const processor = new Processor({ separator: '_' });
    expect(processor.interpret('sm_bg-black').styleSheet.build()).toBe(
      '@media (min-width: 640px) {\n  .sm_bg-black {\n    --tw-bg-opacity: 1;\n    background-color: rgba(0, 0, 0, var(--tw-bg-opacity));\n  }\n}'
    );
  });

  it('add prefix test', () => {
    const processor = new Processor({ prefix: 'tw-' });
    expect(processor.interpret('sm:tw-bg-black').styleSheet.build()).toBe(
      '@media (min-width: 640px) {\n  .sm\\:tw-bg-black {\n    --tw-bg-opacity: 1;\n    background-color: rgba(0, 0, 0, var(--tw-bg-opacity));\n  }\n}'
    );
  });

  it('important test', () => {
    const processor = new Processor({ important: true });
    expect(processor.interpret('sm:bg-black').styleSheet.build()).toBe(
      '@media (min-width: 640px) {\n  .sm\\:bg-black {\n    --tw-bg-opacity: 1 !important;\n    background-color: rgba(0, 0, 0, var(--tw-bg-opacity)) !important;\n  }\n}'
    );
  });

  it('important string test', () => {
    const processor = new Processor({ important: '#app' });
    expect(processor.interpret('sm:bg-black').styleSheet.build()).toBe(
      '@media (min-width: 640px) {\n  #app .sm\\:bg-black {\n    --tw-bg-opacity: 1;\n    background-color: rgba(0, 0, 0, var(--tw-bg-opacity));\n  }\n}'
    );
    expect(processor.interpret('dark:bg-white').styleSheet.build()).toBe(
      '.dark #app .dark\\:bg-white {\n  --tw-bg-opacity: 1;\n  background-color: rgba(255, 255, 255, var(--tw-bg-opacity));\n}'
    );
  });

  it('animation config test', () => {
    const processor = new Processor({
      theme: {
        extend: {
          animation: {
            skbounce: 'skbounce 2.0s infinite ease-in-out',
          },
          keyframes: {
            skbounce: {
              '0%, 100%': { transform: 'scale(0.0)' },
              '50%': { transform: 'scale(1.0)' },
            },
          },
        },
      },
    });
    expect(processor.interpret('animate-skbounce').styleSheet.build()).toMatchSnapshot('animate');
  });

  it('color config test', () => {
    const processor = new Processor({
      darkMode: 'media', // or 'media' or 'class'
      theme: {
        extend: {
          colors: {
            active: 'var(--active)',
            primary: 'var(--primary)',
            'on-primary': 'var(--on-primary)',
            'on-primary-active': 'var(--on-primary-active)',
            frame: 'var(--frame)',
            'on-frame': 'var(--on-frame)',
            'on-frame-active': 'var(--on-frame-active)',
            darkTheme: {
              600: '#262A34',
              700: '#181A20',
              800: '#1A1B20',
            },
          },
        },
      },
    });
    expect(processor.theme('colors.darkTheme')).toEqual({
      600: '#262A34',
      700: '#181A20',
      800: '#1A1B20',
    });

    expect(processor.theme('colors.frame')).toEqual('var(--frame)');

    const styleSheet = processor.interpret('bg-darkTheme-600 bg-active bg-on-primary bg-on-frame-active').styleSheet;
    expect(styleSheet.build()).toMatchSnapshot('css');
  });

  it('handle colors test', () => {
    const processor = new Processor({
      theme: {
        extend: {
          colors: {
            discord: {
              DEFAULT: '#7289da',
              '100': '#7289da',
            },
            'primary-color-red': '#FF0000',
          },
        },
      },
    });
    const styleSheet = processor.interpret(
      'bg-discord bg-discord-100 bg-hex-7289da ring-offset-hex-1c1c1e ring-offset-gray-200 text-primary-color-red'
    ).styleSheet;
    expect(styleSheet.build()).toMatchSnapshot('css');
  });

  it('handle nest colors', () => {
    const processor = new Processor({
      theme: {
        extend: {
          colors: {
            my: {
              DEFAULT: '#000',
              custom: {
                red: '#ff0000',
                DEFAULT: '#fff',
              },
            },
          },
        },
      },
    });

    const styleSheet = processor.interpret('bg-my-custom-red bg-my bg-my-custom text-my-custom').styleSheet;
    expect(styleSheet.build()).toMatchSnapshot('colors');
  });

  it('exclude config', () => {
    const processor = new Processor({
      theme: {
        extend: {
          colors: {
            discord: {
              DEFAULT: '#7289da',
              '100': '#7289da',
            },
          },
        },
      },
      exclude: twExclude,
    });
    expect(processor.interpret('bg-hex-1c1c1e bg-$test-variable p-1rem p-4px p-3 p-4.2 sm:p-4.5').ignored.length).toEqual(7);
  });

  it('shortcuts config string', () => {
    const processor = new Processor({
      shortcuts: {
        'btn': 'py-2 px-4 font-semibold rounded-lg shadow-md',
        'btn-green': 'text-white bg-green-500 hover:bg-green-700',
      },
    });
    expect(processor.interpret('btn btn-green lg:btn').styleSheet.build()).toMatchSnapshot('shortcuts string');
  });

  it('shortcuts config object', () => {
    const processor = new Processor({
      shortcuts: {
        'btn': {
          'color': 'white',
          '@apply': 'py-2 px-4 font-semibold rounded-lg',
          '&:hover': {
            '@apply': 'bg-green-700',
            'color': 'black',
          },
        },
      },
    });
    expect(processor.interpret('btn').styleSheet.build()).toMatchSnapshot('shortcuts object');
  });

  it('allows to use prefix with shortcuts', () => {
    const processor = new Processor({
      prefix: 'windi-',
      shortcuts: {
        'btn': 'py-2 px-4 font-semibold rounded-lg shadow-md',
        'btn-green': {
          '@apply': 'text-white bg-green-500 hover:bg-green-700',
        },
      },
    });
    const result = processor.interpret('windi-btn windi-btn-green');
    expect(result.ignored.length).toEqual(0);
    expect(result.styleSheet.build()).toMatchSnapshot('shortcuts with prefix');
  });

  it('different colors', () => {
    const processor = new Processor({
      theme: {
        colors: {
          name: 'blue',
          hex: '#1c1c1e',
          hexa: '#0000ff00',
          rgb: 'rgb(23, 23, 24)',
          rgba: 'rgba(23, 23, 25, 0.5)',
          hsl: 'hsl(120, 100%, 50%)',
          hsla: 'hsla(120, 100%, 75%, 0.3)',
          hwb: 'hwb(280, 40%, 60%)',
          hwba: 'hwb(280, 40%, 60%, 0)',
          srgb: 'rgb(69.99%, 32%, 32%)',
        },
      },
    });
    expect(processor.interpret('bg-name bg-hex bg-hexa bg-rgb bg-rgba bg-hsl bg-hsla bg-hwb bg-hwba bg-srgb').styleSheet.build()).toMatchSnapshot('colors');
  });
});
