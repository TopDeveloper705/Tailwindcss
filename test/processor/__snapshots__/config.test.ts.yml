Tools / allows to use prefix with shortcuts / shortcuts with prefix / 0: |-
  .windi-btn {
    padding-top: 0.5rem;
    padding-bottom: 0.5rem;
    padding-left: 1rem;
    padding-right: 1rem;
    font-weight: 600;
    border-radius: 0.5rem;
    --tw-shadow: 0 4px 6px -1px rgba(0, 0, 0, 0.1), 0 2px 4px -1px rgba(0, 0, 0, 0.06);
    -webkit-box-shadow: var(--tw-ring-offset-shadow, 0 0 #0000), var(--tw-ring-shadow, 0 0 #0000), var(--tw-shadow);
    box-shadow: var(--tw-ring-offset-shadow, 0 0 #0000), var(--tw-ring-shadow, 0 0 #0000), var(--tw-shadow);
  }
  .windi-btn-green {
    --tw-text-opacity: 1;
    color: rgba(255, 255, 255, var(--tw-text-opacity));
    --tw-bg-opacity: 1;
    background-color: rgba(16, 185, 129, var(--tw-bg-opacity));
  }
  .windi-btn-green:hover {
    --tw-bg-opacity: 1;
    background-color: rgba(4, 120, 87, var(--tw-bg-opacity));
  }
Tools / animation config test / animate / 0: |-
  @keyframes skbounce {
    0%, 100% {
      -webkit-transform: scale(0.0);
      transform: scale(0.0);
    }
    50% {
      -webkit-transform: scale(1.0);
      transform: scale(1.0);
    }
  }
  @-webkit-keyframes skbounce {
    0%, 100% {
      -webkit-transform: scale(0.0);
      transform: scale(0.0);
    }
    50% {
      -webkit-transform: scale(1.0);
      transform: scale(1.0);
    }
  }
  .animate-skbounce {
    -webkit-animation: skbounce 2.0s infinite ease-in-out;
    animation: skbounce 2.0s infinite ease-in-out;
  }
Tools / color config test / css / 0: |-
  .bg-active {
    background-color: var(--active);
  }
  .bg-on-primary {
    background-color: var(--on-primary);
  }
  .bg-on-frame-active {
    background-color: var(--on-frame-active);
  }
  .bg-darkTheme-600 {
    --tw-bg-opacity: 1;
    background-color: rgba(38, 42, 52, var(--tw-bg-opacity));
  }
Tools / config with brackets / css / 0: |-
  .font-bold {
    font-weight: 700;
  }
  .h-\(custom\) {
    height: calc(100% - 4rem);
  }
  .text-lg {
    font-size: 1.125rem;
    line-height: 1.75rem;
  }
Tools / darkColor test / css / 0: |-
  .\~dark\:text-red-500 {
    --tw-text-opacity: 1;
    color: rgba(239, 68, 68, var(--tw-text-opacity));
  }
  @media (prefers-color-scheme: dark) {
    .\~dark\:text-red-500 {
      --tw-text-opacity: 1;
      color: rgba(159, 14, 14, var(--tw-text-opacity));
    }
  }
Tools / darkColor test / css / 1: |-
  .\~dark\:divide-red-200 > :not([hidden]) ~ :not([hidden]) {
    --tw-divide-opacity: 1;
    border-color: rgba(254, 202, 202, var(--tw-divide-opacity));
  }
  .\~dark\:bg-green-300 {
    --tw-bg-opacity: 1;
    background-color: rgba(110, 231, 183, var(--tw-bg-opacity));
  }
  .\~dark\:placeholder-gray-200::placeholder {
    --tw-placeholder-opacity: 1;
    color: rgba(28, 28, 30, var(--tw-placeholder-opacity));
  }
  .\~dark\:text-blue-400 {
    --tw-text-opacity: 1;
    color: rgba(51, 154, 240, var(--tw-text-opacity));
  }
  @media (prefers-color-scheme: dark) {
    .\~dark\:divide-red-200 > :not([hidden]) ~ :not([hidden]) {
      --tw-divide-opacity: 1;
      border-color: rgba(82, 2, 2, var(--tw-divide-opacity));
    }
    .\~dark\:bg-green-300 {
      --tw-bg-opacity: 1;
      background-color: rgba(22, 131, 98, var(--tw-bg-opacity));
    }
    .\~dark\:text-blue-400 {
      --tw-text-opacity: 1;
      color: rgba(165, 216, 255, var(--tw-text-opacity));
    }
  }
Tools / dashify / css / 0: |-
  .foo {
    --tw-bg-opacity: 1;
  }
  .bar {
    tw-bg-opacity: 1;
  }
  .caps {
    test-css: 1;
  }
Tools / different colors / colors / 0: |-
  .bg-name {
    --tw-bg-opacity: 1;
    background-color: rgba(0, 0, 255, var(--tw-bg-opacity));
  }
  .bg-hex {
    --tw-bg-opacity: 1;
    background-color: rgba(28, 28, 30, var(--tw-bg-opacity));
  }
  .bg-hexa {
    --tw-bg-opacity: 0;
    background-color: rgba(0, 0, 255, var(--tw-bg-opacity));
  }
  .bg-rgb {
    --tw-bg-opacity: 1;
    background-color: rgba(23, 23, 24, var(--tw-bg-opacity));
  }
  .bg-rgba {
    --tw-bg-opacity: 0.5;
    background-color: rgba(23, 23, 25, var(--tw-bg-opacity));
  }
  .bg-hsl {
    --tw-bg-opacity: 1;
    background-color: rgba(0, 255, 0, var(--tw-bg-opacity));
  }
  .bg-hsla {
    --tw-bg-opacity: 0.3;
    background-color: rgba(128, 255, 128, var(--tw-bg-opacity));
  }
  .bg-hwb {
    --tw-bg-opacity: 1;
    background-color: rgba(102, 102, 102, var(--tw-bg-opacity));
  }
  .bg-hwba {
    --tw-bg-opacity: 0.7;
    background-color: rgba(102, 102, 102, var(--tw-bg-opacity));
  }
  .bg-srgb {
    --tw-bg-opacity: 1;
    background-color: rgba(178, 82, 82, var(--tw-bg-opacity));
  }
Tools / extend black / css / 0: |-
  .bg-black-300 {
    --tw-bg-opacity: 1;
    background-color: rgba(28, 28, 30, var(--tw-bg-opacity));
  }
  .bg-black {
    --tw-bg-opacity: 1;
    background-color: rgba(0, 0, 0, var(--tw-bg-opacity));
  }
  .bg-black-400 {
    --tw-bg-opacity: 1;
    background-color: rgba(34, 0, 0, var(--tw-bg-opacity));
  }
Tools / handle colors test / css / 0: |-
  .bg-discord {
    --tw-bg-opacity: 1;
    background-color: rgba(114, 137, 218, var(--tw-bg-opacity));
  }
  .bg-discord-100 {
    --tw-bg-opacity: 1;
    background-color: rgba(114, 137, 218, var(--tw-bg-opacity));
  }
  .bg-hex-7289da {
    --tw-bg-opacity: 1;
    background-color: rgba(114, 137, 218, var(--tw-bg-opacity));
  }
  .ring-offset-hex-1c1c1e {
    --tw-ring-offset-color: #1c1c1e;
  }
  .ring-offset-gray-200 {
    --tw-ring-offset-color: #e5e7eb;
  }
  .text-primary-color-red {
    --tw-text-opacity: 1;
    color: rgba(255, 0, 0, var(--tw-text-opacity));
  }
Tools / handle nest colors / colors / 0: |-
  .bg-my-custom-red {
    --tw-bg-opacity: 1;
    background-color: rgba(255, 0, 0, var(--tw-bg-opacity));
  }
  .bg-my {
    --tw-bg-opacity: 1;
    background-color: rgba(0, 0, 0, var(--tw-bg-opacity));
  }
  .bg-my-custom {
    --tw-bg-opacity: 1;
    background-color: rgba(255, 255, 255, var(--tw-bg-opacity));
  }
  .text-my-custom {
    --tw-text-opacity: 1;
    color: rgba(255, 255, 255, var(--tw-text-opacity));
  }
Tools / important string test / css / 0: |-
  @media (min-width: 640px) {
    #app .sm\:bg-black {
      --tw-bg-opacity: 1;
      background-color: rgba(0, 0, 0, var(--tw-bg-opacity));
    }
  }
Tools / important string test / css / 1: |-
  .dark #app .dark\:bg-white {
    --tw-bg-opacity: 1;
    background-color: rgba(255, 255, 255, var(--tw-bg-opacity));
  }
Tools / resolveConfig should extend correctly / order / 0: |-
  {
    "1": "1",
    "2": "2",
    "3": "3",
    "4": "4",
    "5": "5",
    "6": "6",
    "7": "7",
    "8": "8",
    "9": "9",
    "10": "10",
    "11": "11",
    "12": "12",
    "first": "-9999",
    "last": "9999",
    "none": "0",
    "lg": "44"
  }
Tools / resolveConfig should not overwrite theme function / extend-colors-red / 1: |-
  {
    "50": [
      "#fef2f2",
      "#370404"
    ],
    "100": [
      "#fee2e2",
      "#430202"
    ],
    "200": [
      "#fecaca",
      "#520202"
    ],
    "300": [
      "#fca5a5",
      "#670303"
    ],
    "400": [
      "#f87171",
      "#860707"
    ],
    "500": [
      "#ef4444",
      "#9f0e0e"
    ],
    "600": [
      "#dc2626",
      "#ae1c1c"
    ],
    "700": [
      "#b91c1c",
      "#941616"
    ],
    "800": [
      "#991b1b",
      "#7a1616"
    ],
    "900": [
      "#7f1d1d",
      "#661717"
    ],
    "custom": "#ea1851"
  }
Tools / resolveConfig should not overwrite theme function / extend-width / 0: |-
  {
    "0": "0px",
    "1": "0.25rem",
    "2": "0.5rem",
    "3": "0.75rem",
    "4": "1rem",
    "5": "1.25rem",
    "6": "1.5rem",
    "7": "1.75rem",
    "8": "2rem",
    "9": "2.25rem",
    "10": "2.5rem",
    "11": "2.75rem",
    "12": "3rem",
    "14": "3.5rem",
    "16": "4rem",
    "20": "5rem",
    "24": "6rem",
    "28": "7rem",
    "32": "8rem",
    "36": "9rem",
    "40": "10rem",
    "44": "11rem",
    "48": "12rem",
    "52": "13rem",
    "56": "14rem",
    "60": "15rem",
    "64": "16rem",
    "72": "18rem",
    "80": "20rem",
    "96": "24rem",
    "auto": "auto",
    "px": "1px",
    "0.5": "0.125rem",
    "1.5": "0.375rem",
    "2.5": "0.625rem",
    "3.5": "0.875rem",
    "1/2": "50%",
    "1/3": "33.333333%",
    "2/3": "66.666667%",
    "1/4": "25%",
    "2/4": "50%",
    "3/4": "75%",
    "1/5": "20%",
    "2/5": "40%",
    "3/5": "60%",
    "4/5": "80%",
    "1/6": "16.666667%",
    "2/6": "33.333333%",
    "3/6": "50%",
    "4/6": "66.666667%",
    "5/6": "83.333333%",
    "1/12": "8.333333%",
    "2/12": "16.666667%",
    "3/12": "25%",
    "4/12": "33.333333%",
    "5/12": "41.666667%",
    "6/12": "50%",
    "7/12": "58.333333%",
    "8/12": "66.666667%",
    "9/12": "75%",
    "10/12": "83.333333%",
    "11/12": "91.666667%",
    "xs": "20rem",
    "sm": "24rem",
    "md": "28rem",
    "lg": "32rem",
    "xl": "36rem",
    "2xl": "42rem",
    "3xl": "48rem",
    "4xl": "56rem",
    "5xl": "64rem",
    "6xl": "72rem",
    "7xl": "80rem",
    "full": "100%",
    "min": "min-content",
    "max": "max-content",
    "prose": "65ch",
    "screen": "100vw",
    "screen-sm": "640px",
    "screen-md": "768px",
    "screen-lg": "1024px",
    "screen-xl": "1280px",
    "screen-2xl": "1536px",
    "1/7": "14%",
    "2/7": "28%",
    "3/7": "42%",
    "4/7": "57%",
    "5/7": "71%",
    "6/7": "85%"
  }
Tools / shortcuts config object / shortcuts object / 0: |-
  .btn {
    color: white;
    padding-top: 0.5rem;
    padding-bottom: 0.5rem;
    padding-left: 1rem;
    padding-right: 1rem;
    font-weight: 600;
    border-radius: 0.5rem;
  }
  .btn:hover {
    --tw-bg-opacity: 1;
    background-color: rgba(4, 120, 87, var(--tw-bg-opacity));
    color: black;
  }
Tools / shortcuts config string / shortcuts string / 0: |-
  .btn {
    padding-top: 0.5rem;
    padding-bottom: 0.5rem;
    padding-left: 1rem;
    padding-right: 1rem;
    font-weight: 600;
    border-radius: 0.5rem;
    --tw-shadow: 0 4px 6px -1px rgba(0, 0, 0, 0.1), 0 2px 4px -1px rgba(0, 0, 0, 0.06);
    -webkit-box-shadow: var(--tw-ring-offset-shadow, 0 0 #0000), var(--tw-ring-shadow, 0 0 #0000), var(--tw-shadow);
    box-shadow: var(--tw-ring-offset-shadow, 0 0 #0000), var(--tw-ring-shadow, 0 0 #0000), var(--tw-shadow);
  }
  .btn-green {
    --tw-text-opacity: 1;
    color: rgba(255, 255, 255, var(--tw-text-opacity));
    --tw-bg-opacity: 1;
    background-color: rgba(16, 185, 129, var(--tw-bg-opacity));
  }
  .btn-green:hover {
    --tw-bg-opacity: 1;
    background-color: rgba(4, 120, 87, var(--tw-bg-opacity));
  }
  @media (min-width: 1024px) {
    .lg\:btn {
      padding-top: 0.5rem;
      padding-bottom: 0.5rem;
      padding-left: 1rem;
      padding-right: 1rem;
      font-weight: 600;
      border-radius: 0.5rem;
      --tw-shadow: 0 4px 6px -1px rgba(0, 0, 0, 0.1), 0 2px 4px -1px rgba(0, 0, 0, 0.06);
      -webkit-box-shadow: var(--tw-ring-offset-shadow, 0 0 #0000), var(--tw-ring-shadow, 0 0 #0000), var(--tw-shadow);
      box-shadow: var(--tw-ring-offset-shadow, 0 0 #0000), var(--tw-ring-shadow, 0 0 #0000), var(--tw-shadow);
    }
  }
Tools / variables theme config / css / 0: |-
  :root {
    --primary: #6366f1;
  }
